### PR TITLE
ci: improve artifact stuff in workflows

### PR DESCRIPTION
**Pull request by ylobankov**
Closes tarantool/tarantool-qa#156

### CODE REVIEW

File: .github/workflows/freebsd-13.yml
Comment (Totktonada) [inline .github/workflows/freebsd-13.yml:58]: See c820754389eb0764c337b1e39b44ab0074dbb47b.

Comment (ylobankov) [reply .github/workflows/freebsd-13.yml:58]: Looks like this limitation is gone. See this [section](https://github.com/ChristopherHX/github-act-runner/tree/58ae37abc6c2244d91822b8ba536aa0a8b829632#known-limitations) in README.md. But likely we need to update our agents on freebsd runners.

Comment (ylobankov) [reply .github/workflows/freebsd-13.yml:58]: Agents are updated to `v0.2.0` version where this limitation is not relevant anymore.

Comment (ylobankov) [reply .github/workflows/freebsd-13.yml:58]: Artifact creation is tested, works fine.

File: .github/workflows/debug_aarch64.yml
Comment (Totktonada) [inline .github/workflows/debug_aarch64.yml:67]: 1. It does not look related to the commit 'ci: fix hardcoded artifact paths in workflows'.
2. AFAIU we can't set the same name for artifacts from different workflows, because they will rewrite each other. That's why they was named similarly to a workflow name.

(The same for the artifact name in `coverage.yml`.)

File: .github/actions/environment/action.yml
Comment (Totktonada) [inline .github/actions/environment/action.yml:12]: Nit: I would add that it follows up f1d750f036956873c79bf5de51f77b38da25ae0f.

File: .github/actions/environment/action.yml
Comment (Totktonada) [inline .github/actions/environment/action.yml:85]:
```markdown
# Whether to execute `git clean -ffdx && git reset --hard HEAD` before fetching
# Default: true
clean: ''
```

https://github.com/actions/checkout/blob/main/README.md

It seems, it should be performed by the checkout action.

**Review by Totktonada — APPROVED** [review]
No objections in general (but beware: I just glanced over the PR and didn't test changes). I left several comments. I'm okay to push the PR after fixes and I don't think that it requires second review from me.

**Review by kyukhin — APPROVED** [review]

File: .github/workflows/debug_aarch64.yml
Comment (ylobankov) [reply .github/workflows/debug_aarch64.yml:67]: It looks like the second item is relevant to workflows with a few jobs inside that upload artifacts [1], but not related to jobs running in different workflows. Uploading to the same artifact multiple times in the same workflow can overwrite the artifact.

[1] https://github.com/actions/upload-artifact/tree/v2#uploading-to-the-same-artifact

File: .github/actions/environment/action.yml
Comment (ylobankov) [reply .github/actions/environment/action.yml:85]: Yes, now I see it. But I think it is not bad that we also explicitly run this command.

File: .github/actions/environment/action.yml
Comment (ylobankov) [reply .github/actions/environment/action.yml:12]: Sorry, I was late.

File: .github/workflows/debug_aarch64.yml
Comment (Totktonada) [reply .github/workflows/debug_aarch64.yml:67]: TBH, I don't know for sure. I would appreciate if you'll monitor how it behave during some time.

File: .github/actions/environment/action.yml
Comment (Totktonada) [reply .github/actions/environment/action.yml:85]: I don't mind anyway.

File: .github/workflows/debug_aarch64.yml
Comment (ylobankov) [reply .github/workflows/debug_aarch64.yml:67]: Ok, will do.

**Pull request merged**
